ToolChain/Assembler/Parser/test_Parser.py

```diff
@@ -37,978 +35,1368 @@
+
+        instruction = self.parser._findInstructionCode("Ins", ["HIRET"], b'')
+        correctInstruction = 0b11110011
+        self.assertEqual(instruction, bytes((correctInstruction,)))
+
+        instruction = self.parser._findInstructionCode("Ins", ["NOP"], b'')
+        correctInstruction = 0b11111111
+        self.assertEqual(instruction, bytes((correctInstruction,)))
+
+        instruction = self.parser._findInstructionCode("Ins", ["RET"], b'')
+        correctInstruction = 0b11110000
+        self.assertEqual(instruction, bytes((correctInstruction,)))
+
+    def test_findInstructionCodeInsError(self):
+        try:
+            self.parser._findInstructionCode("Ins", ["error"], b'')
+        except ValueError:
+            pass
+        else:
+            raise AssertionError("ValueError was not raised")
+
+        try:
+            self.parser._findInstructionCode("InsReg", ["RET"], b'')
+        except ValueError:
+            pass
+        else:
+            raise AssertionError("ValueError was not raised")
+
+    def test_findInstructionCodeInsReg(self):
+        """
+        Tests the method:
+        _findInstructionCode(self, form, line, instruction)
+        for instruction in STATE1 (InsReg)
+        """
+
+        instruction = self.parser._findInstructionCode("InsReg", ["CALL", "$B"], b'')
+        correctInstruction = 0b01110010
+        self.assertEqual(instruction, bytes((correctInstruction,)))
+
+        instruction = self.parser._findInstructionCode("InsReg", ["INT", "$B"], b'')
+        correctInstruction = 0b01110110
+        self.assertEqual(instruction, bytes((correctInstruction,)))
+
+        instruction = self.parser._findInstructionCode("InsReg", ["NOT", "$B"], b'')
+        correctInstruction = 0b01110000
+        self.assertEqual(instruction, bytes((correctInstruction,)))
+
+        instruction = self.parser._findInstructionCode("InsReg", ["POP", "$C"], b'')
+        correctInstruction = 0b01110100
+        self.assertEqual(instruction, bytes((correctInstruction,)))
+
+        instruction = self.parser._findInstructionCode("InsReg", ["PUSH", "$C"], b'')
+        correctInstruction = 0b01110011
+        self.assertEqual(instruction, bytes((correctInstruction,)))
+
+        instruction = self.parser._findInstructionCode("InsReg", ["SIVR", "$B"], b'')
+        correctInstruction = 0b01110101
+        self.assertEqual(instruction, bytes((correctInstruction,)))
+
+    def test_findInstructionCodeInsRegError(self):
+
+        try:
+            self.parser._findInstructionCode("InsReg", ["error"], b'')
+        except ValueError:
+            pass
+        else:
+            raise AssertionError("ValueError was not raised")
+
+        try:
+            self.parser._findInstructionCode("Ins", ["CALL"], b'')
+        except ValueError:
+            pass
+        else:
+            raise AssertionError("ValueError was not raised")
 
-    def test_init(self):
+    def test_findInstructionCodeInsRegReg(self):
```

Comment:
Since tests are sort of their own thing, I wouldn't mind this naming convention. Though I think @pg-cse wanted consistency throughout the whole project.